PrivateClinicsWebNet.Application/Services/AppointmentService.cs

```diff
@@ -0,0 +1,99 @@
+            bool isExist = await _appointmentRepository.IsAppointmentExistAsync(appointment.Id.ToString());
+            if (!isExist)
+                return Result<string>.Failure($"The appointment with ID {appointment.Id.ToString()} cannot be update." +
+                    $"It doesn`t exist in system yet");
+            await _appointmentRepository.UpdateAppointmentAsync(appointment);
+            return await Result<string>.SuccessAsync("Appointment was updated successfully.");
+        }
+
+        public async Task<Result<List<AppointmentResponseDto>>> GetDoctorAppointmentsAsync(string id, AppointmentsRequestDto requestDto)
+        {
+            Expression<Func<Appointment, bool>> matchesDoctorId = a => a.DoctorId == id;
+            var appointmentsList = await _appointmentRepository.GetUserAppointmentsAsync(matchesDoctorId, requestDto.Page, requestDto.PageSize);
+            if (appointmentsList == null)
+                return Result<List<AppointmentResponseDto>>.Failure($"Doctor with ID {id} doesn`t have any appointments yet.");
+            var appointmentListResponse = new List<AppointmentResponseDto>();
+            foreach (var appointment in appointmentsList)
+            {
+                var response = _mapper.Map<AppointmentResponseDto>(appointment);
+                appointmentListResponse.Add(response);
+            }
+            return await Result<List<AppointmentResponseDto>>.SuccessAsync(appointmentListResponse);
+        }
+
+        public async Task<Result<List<AppointmentResponseDto>>> GetPatientAppointmentsAsync(string id, AppointmentsRequestDto requestDto)
+        {
+            Expression<Func<Appointment, bool>> matchesDoctorId = a => a.PatientId == id;
+            var appointmentsList = await _appointmentRepository.GetUserAppointmentsAsync(matchesDoctorId, requestDto.Page, requestDto.PageSize);
+            if (appointmentsList == null)
+                return Result<List<AppointmentResponseDto>>.Failure($"Patient with ID {id} doesn`t have any appointments yet.");
+            var appointmentListResponse = new List<AppointmentResponseDto>();
+            foreach (var appointment in appointmentsList)
+            {
+                var response = _mapper.Map<AppointmentResponseDto>(appointment);
+                appointmentListResponse.Add(response);
+            }
+            return await Result<List<AppointmentResponseDto>>.SuccessAsync(appointmentListResponse);
+        }
+
+        public async Task<Result<string>> DeleteAppointmentAsync(string appointmentId)
+        {
+            bool isExist = await _appointmentRepository.IsAppointmentExistAsync(appointmentId);
+            if (!isExist)
+                return Result<string>.Failure($"Appointment with ID {appointmentId} doesn`t exist in the system");
+            await _appointmentRepository.DeleteAppointmentAsync(appointmentId);
+            bool hasBeenDeleted = !await _appointmentRepository.IsAppointmentExistAsync(appointmentId);
```

Comment:
Why we have 3 queries instead of 1 query to delete? Optimize please your queries here and all other queries